src/docfx/lib/UrlUtility.cs

```diff
@@ -344,6 +344,68 @@ public static string RemoveLeadingHostName(string url, string hostName, bool rem
             : $"/{path}";
     }
 
+    public static string GetBookmark(string uid)
+    {
+        uid = uid.ToLowerInvariant();
+        var shouldIgnore = new HashSet<char>() { '\"', '\'', '%', '^', '\\' };
```

Comment:
The `uid.ToLowerInvariant()` call can be done at char level and moved inside the for loop.

Reply:
Removed.